src/hyperion/external_interaction/nexus/write_nexus.py

```diff
@@ -13,68 +13,51 @@
 from nexgen.nxs_utils import Attenuator, Beam, Detector, Goniometer, Source
 from nexgen.nxs_write.nxmx_writer import NXmxFileWriter
 from numpy.typing import DTypeLike
+from scanspec.core import AxesPoints
 
 from hyperion.external_interaction.nexus.nexus_utils import (
     create_detector_parameters,
     create_goniometer_axes,
     get_start_and_predicted_end_time,
 )
-from hyperion.parameters.internal_parameters import (
-    HyperionParameters,
-    InternalParameters,
-)
+from hyperion.parameters.gridscan import ThreeDGridScan
+from hyperion.parameters.rotation import RotationScan
 
 
 class NexusWriter:
     def __init__(
         self,
-        parameters: InternalParameters,
-        scan_points: dict,
+        parameters: ThreeDGridScan | RotationScan,
         data_shape: tuple[int, int, int],
-        omega_start: float | None = None,
+        *,
+        scan_points: AxesPoints | None = None,
         run_number: int | None = None,
+        omega_start_deg: float | None = None,
         vds_start_index: int = 0,
     ) -> None:
         self.beam: Optional[Beam] = None
         self.attenuator: Optional[Attenuator] = None
-        self.scan_points: dict = scan_points
+        self.scan_points: dict = scan_points or parameters.scan_points
```

Comment:
Not sure I'm really a fan of sometimes-we-initialise-from-parameters, sometimes we don't, I think it makes things less clear at the call site.